app/src/main/java/ingsw/group1/findmyphone/fragment/SettingsFragment.java

```diff
@@ -0,0 +1,27 @@
+package ingsw.group1.findmyphone.fragment;
+
+import android.os.Bundle;
+
+import androidx.preference.PreferenceFragmentCompat;
+
+import ingsw.group1.findmyphone.R;
+
+/**
+ * Fragment dedicated to showing the Preferences for the app.
+ *
+ * @author Riccardo De Zen
+ */
+public class SettingsFragment extends PreferenceFragmentCompat {
+
+    /**
+     * Method called to inflate the Preferences hierarchy.
+     *
+     * @param savedInstanceState The saved instance state.
+     * @param rootKey            The root key, for the hierarchy, it is provided automatically.
+     */
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.app_preferences, rootKey);
```

Comment:
The root key will the password for encrypting outgoing messages? (I've fallen a little behind the logic)

Reply:
Not really important in this case, a `PreferenceScreen` in a preferences xml file can have a certain key, if a certain screen key is provided to this method then the fragment should use it as root for the preferences to show. I think when called by the system will be null by default meaning to use the default screen in a file, which is our case. I wanted to put it always null but it didn't seem to hurt like this

Reply:
Also this should call the superclass method not sure why I have forgot

Reply:
![image](https://user-images.githubusercontent.com/56435959/74881455-a4d22e80-536d-11ea-93e0-73d1039583f3.png)

Reply:
Ah yes I see you got my not so subtle message

Reply:
Gotta try that game someday

Reply:
* honk *

Reply:
> Not really important in this case, a `PreferenceScreen` in a preferences XML file can have a certain key, if a certain screen key is provided to this method then the fragment should use it as root for the preferences to show. I think when called by the system will be null by default meaning to use the default screen in a file, which is our case. I wanted to put it always null but it didn't seem to hurt like this

Now, being serious, I kinda get it 👌